modules/component-store/spec/component-store.spec.ts

```diff
@@ -401,4 +401,199 @@ describe('Component Store', () => {
       })
     );
   });
+
+  describe('selectors', () => {
+    interface State {
+      value: string;
+      updated?: boolean;
+    }
+
+    const INIT_STATE: State = { value: 'init' };
+    let componentStore: ComponentStore<State>;
+
+    beforeEach(() => {
+      componentStore = new ComponentStore<State>(INIT_STATE);
+    });
+
+    it(
+      'uninitialized Component Store does not emit values',
+      marbles(m => {
+        const uninitializedComponentStore = new ComponentStore();
+        m.expect(uninitializedComponentStore.select(s => s)).toBeObservable(
+          m.hot('-')
+        );
+      })
+    );
+
+    it(
+      'selects component root state',
+      marbles(m => {
+        m.expect(componentStore.select(s => s)).toBeObservable(
+          m.hot('i', { i: INIT_STATE })
+        );
+      })
+    );
+
+    it(
+      'selects component property from the state',
+      marbles(m => {
+        m.expect(componentStore.select(s => s.value)).toBeObservable(
+          m.hot('i', { i: INIT_STATE.value })
+        );
+      })
+    );
+
+    it(
```

Comment:
I'd like to see one more test that shows the max number of selectors, and still only one emission even if all 4 emit an update. Could be a follow-up

Reply:
Done